web/src/message_list_view.js

```diff
@@ -525,8 +524,6 @@ export class MessageListView {
         };
 
         for (const message_container of message_containers) {
-            const message_reactions = reactions.get_message_reactions(message_container.msg);
```

Comment:
This first commit moves where we process reactions from when we render the message to when we process the message for the first time, which I believe means we won't be updating the reactions as often as we did before. But there's this block that suggests to me that we do want to update them each time we render:

```js
    if (message.clean_reactions) {
        // Update display details for the reaction. In particular,
        // user_settings.display_emoji_reaction_users or the names of
        // the users appearing in the reaction may have changed since
        // this reaction was first rendered.
        const reaction_counts_and_user_ids = get_reaction_counts_and_user_ids(message);
        const should_display_reactors = check_should_display_reactors(reaction_counts_and_user_ids);
        for (const clean_reaction of message.clean_reactions.values()) {
            update_user_fields(clean_reaction, should_display_reactors);
        }
        return;
    }
```

Reply:
Also the commit message has a typo in it: "meassge"

Reply:
Ah okay, this issue has already been realized https://github.com/zulip/zulip/commit/3df46d542bc3cd9f696ec6002e37e79009bd3e55

Reply:
Ok, so should I process the `message_reactions` when we process new messages as well as everytime we render?

Also I still don't understand why we're not updating `message_reactions` when adding a new reaction?

Reply:
> But there's this block that suggests to me that we do want to update them each time we render.

This block updates the user fields (reaction count, etc), but i don't think we need to call this block every time we render because we are anyways updating it every time we add/remove/modify any reaction.